examples/seq2seq/finetune_trainer.py

```diff
@@ -106,6 +107,46 @@ class DataTrainingArguments:
     )
 
 
+def speed_metrics(mode, t0, n_objs):
+    """
+    Measure and return speed performance metrics.
+
+    This function requires a time snapshot `t0` before the operation to be measured starts and this
+    function should be run immediately after the operation to be measured has completed.
+
+    Args:
+    - mode: one of train, eval, test
+    - t0: operation start time
+    - n_objs: number of samples processed
+
+    """
+    runtime = time.time() - t0
+    result = {}
+
+    samples_per_second = 1 / (runtime / n_objs)
+    result[f"{mode}_samples_per_second"] = round(samples_per_second, 3)
+    result[f"{mode}_runtime"] = round(runtime, 4)
+
+    result[f"{mode}_n_ojbs"] = n_objs
+    return result
+
+
+def handle_metrics(mode, metrics, output_dir):
+    """
+    Log and save metrics
+
+    Args:
+    - mode: one of train, eval, test
+    - metrics: metrics dict
+    - output_dir: where to save the metrics
+    """
+
+    logger.info(f"***** {mode} metrics *****")
+    for key, value in metrics.items():
+        logger.info("  %s = %s", key, value)
```

Comment:
```suggestion
        logger.info(f"  {key} = {value}")
```
Let's use f-strings pretty please.